### PR TITLE
fix build script and test script, add github action to automate tests

### LINKED ISSUE #2: Project cannot be built
The project uses TypeScript to build

https://github.com/holocron-hq/safe-mdx/blob/66aa3dbf12d4254dfa1b622ff7524ddb3928339c/package.json#L19

but TypeScript is not included in the dependencies

https://github.com/holocron-hq/safe-mdx/blob/66aa3dbf12d4254dfa1b622ff7524ddb3928339c/package.json#L35-L52

TypeScript should be included as a `devDependency` to track the version used to build.

In addition, the TypeScript configuration file is incomplete, it references a relative file outside the repository

https://github.com/holocron-hq/safe-mdx/blob/66aa3dbf12d4254dfa1b622ff7524ddb3928339c/tsconfig.json#L2

When building with TypeScript 5 this results in the error:
```
error TS5083: Cannot read file '../tsconfig.base.json'.
```

### LINKED ISSUE #3: Project tests cannot be run
The project leverage `vitest` for testing

https://github.com/holocron-hq/safe-mdx/blob/66aa3dbf12d4254dfa1b622ff7524ddb3928339c/package.json#L20

but `vitest` is not included as a `devDependency`

https://github.com/holocron-hq/safe-mdx/blob/66aa3dbf12d4254dfa1b622ff7524ddb3928339c/package.json#L35-L52

no vitest configuration is included to set how the test should run

### PR DESCRIPTION
**Pull request by ChristianMurphy**
resolves #2
resolves #3

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
-    "extends": "../tsconfig.base.json",
     "compilerOptions": {
-        "rootDir": "src",
+        "target": "es2022",
         "jsx": "react-jsx",
-        "module": "ES2015",
-        "outDir": "dist"
+        "module": "node16",
+        "outDir": "dist",
+        "strict": false,
```
Comment (ChristianMurphy) [inline tsconfig.json:7]: this should be enabled in the future

File: package.json [hunk 2]
```diff
@@ -33,21 +33,23 @@
         "react": "*"
     },
     "dependencies": {
-        "html-to-jsx-transform": "^1.1.0",
-        "mdast": "^3.0.0",
```
Comment (ChristianMurphy) [inline package.json:37]: mdast is a types only dependency.
There was an implementation many years ago, it shouldn't be used or installed.

File: package.json [hunk 3]
```diff
@@ -33,21 +33,23 @@
...
-        "remark": "^15.0.1",
+        "html-to-jsx-transform": "^1.0.0",
+        "remark": "^15.0.0",
```
Comment (ChristianMurphy) [inline package.json:37]: where possible use wide version ranges to allow downstream packages to deduplicate as much as possible

File: tsconfig.json [hunk 4]
```diff
@@ -1,10 +1,13 @@
 {
-    "extends": "../tsconfig.base.json",
     "compilerOptions": {
-        "rootDir": "src",
+        "target": "es2022",
         "jsx": "react-jsx",
-        "module": "ES2015",
-        "outDir": "dist"
+        "module": "node16",
```
Comment (ChristianMurphy) [inline tsconfig.json:5]: > node16 and nodenext are the only correct module options for all apps and libraries that are intended to run in Node.js v12 or later, whether they use ES modules or not.

https://www.typescriptlang.org/docs/handbook/modules/reference.html#summary

File: .github/workflows/main.yml
Comment (ChristianMurphy) [inline .github/workflows/main.yml:1]: this action enables automated validation of the build and test suite

**Comment by ChristianMurphy** [comment]
Here's an example of the CI/CD output https://github.com/ChristianMurphy/safe-mdx/actions/runs/7910959769

**Comment by remorses** [comment]
thank you

**Pull request merged**